### PR TITLE
Add importing from a data slice to the public importer API

### PR DESCRIPTION
**Pull request by Rhuagh**

### CODE REVIEW

**Comment by alteous** [comment]
Thanks for the PR!

You're merging this with my personal fork. It might be a better idea to merge into the `incoming` branch of `https://github.com/gltf-rs/gltf.git` instead.

File: gltf-importer/src/lib.rs [hunk 1]
```diff
@@ -107,12 +107,16 @@ impl Buffers {
 
 fn import_impl(path: &Path, config: Config) -> Result<(Gltf, Buffers), Error> {
     let data = read_to_end(path)?;
+    import_data_slice(&data, path, config)
+}
+
+pub fn import_data_slice<'a>(data: &'a [u8], path: &Path, config: Config) -> Result<(Gltf, Buffers), Error> {
```
Comment (alteous) [inline gltf-importer/src/lib.rs:113]: Missing doc comment and `AsRef<Path>` but it's no big deal.

**Review by alteous — APPROVED** [review]

**Comment by alteous** [comment]
Would you prefer for me to merge the PR into my repo, make some changes, and merge into the upstream `gltf-rs/gltf`, or to merge into `gltf-rs/gltf` directly?

File: gltf-importer/src/lib.rs (hunk 1 above)
Comment (alteous) [reply gltf-importer/src/lib.rs:113]: We use `import_impl` to keep the compile time sane, FYI.

**Comment by Rhuagh** [comment]
Do that, thanks!

**Comment by Rhuagh** [comment]
And your branch is the first hit on Google :P

**Comment by alteous** [comment]
Okay, thank you. :smile:

**Pull request merged**